static/app/views/performance/transactionSummary/transactionEvents/content.spec.tsx

```diff
@@ -228,25 +216,20 @@ describe('Performance Transaction Events Content', function () {
           />
         </OrganizationContext.Provider>
       </RouteContext.Provider>,
-      initialData.routerContext
+      {context: initialData.routerContext}
     );
-    await tick();
-    wrapper.update();
 
-    expect(wrapper.find('EventsTable')).toHaveLength(1);
-    expect(wrapper.find('CompactSelect')).toHaveLength(1);
-    expect(wrapper.find('StyledSearchBar')).toHaveLength(1);
-    expect(wrapper.find('Filter')).toHaveLength(1);
+    expect(screen.getByTestId('events-table')).toBeInTheDocument();
+    expect(screen.getByText(textWithMarkupMatcher('Percentilep100'))).toBeInTheDocument();
```

Comment:
is `Percentilep100` rendered if the `events-table` is not rendered? do we need to check for both?

Reply:
All the components are unconditionally included, but some of them will return `null`. For example `<Filter>` will guard on the feature `performance-ops-breakdown`.

I think the tests before are weaker because the react components are always in the document. So they couldn't really fail unless the return statements changed.
Now they're a little better because we're looking at some kind of rendered content. But lmk if you think it could go further.

https://github.com/getsentry/sentry/blob/705c27aa41733ff3b88d6da93fcf3a9d86cd190f/static/app/views/performance/transactionSummary/transactionEvents/content.tsx#L86-L98

https://github.com/getsentry/sentry/blob/705c27aa41733ff3b88d6da93fcf3a9d86cd190f/static/app/views/performance/transactionSummary/transactionEvents/content.tsx#L143-L169